### PR TITLE
SAK-43470 TurnitinOC: Pass TCA user roles when submitting and viewing…

### PR DESCRIPTION
**Pull request by baholladay**
… reports

### CODE REVIEW

File: content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java [hunk 1]
```diff
@@ -159,8 +166,26 @@
 	private static final String VIEWER_PERMISSION_MAY_VIEW_SUBMISSIONS_FULL_SOURCE = "may_view_submission_full_source";
 	private static final String VIEWER_PERMISSION_MAY_VIEW_MATCH_SUBMISSION_INFO = "may_view_match_submission_info";
 	private static final String VIEWER_DEFAULT_PERMISSIONS = "viewer_default_permission_set";
-	private static final String INSTRUCTOR = "INSTRUCTOR";
-	private static final String LEARNER = "LEARNER";
+	private enum ROLES{
+		INSTRUCTOR(Arrays.asList("Faculty", "Instructor", "Mentor", "Staff", "maintain", "Teaching Assistant")),
```
Comment (baholladay) [inline content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: I just looked through the existing default templates... are there any default Sakai roles that I may have missed? Custom roles can be mapped in the sakai.properties?

File: content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java [hunk 2]
```diff
@@ -883,8 +911,14 @@ private String getSubmissionId(ContentReviewItem item, String fileName, Site sit
...
 			Map<String, Object> data = new HashMap<String, Object>();
 			data.put("owner", userID);
+			if(site != null) {
+				data.put("owner_default_permission_set", mapUserRole(userID, site.getId(), false));
+			}
 			if (StringUtils.isNotBlank(submitterID)) {
-				data.put("submitter", submitterID);	
+				data.put("submitter", submitterID);
+				if(site != null) {
+					data.put("submitter_default_permission_set", mapUserRole(submitterID, site.getId(), false));
```
Comment (bjones86) [inline content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java]: Can this be put into the above `if(site != null)` block? Conditional seems duplicated.

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5236832&bid=17165239)

**Comment by steveswinsburg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/sakaiproject/sakai/pullRequest?prid=5236832&bid=17165239)

**Comment by baholladay** [comment]
@ern is it ok to merge?

**Pull request merged**